src/cargo/sources/path.rs

```diff
@@ -162,50 +175,59 @@ impl<'cfg> PathSource<'cfg> {
         &self,
         pkg: &Package,
         root: &Path,
-        filter: &mut dyn FnMut(&Path) -> CargoResult<bool>,
+        filter: &mut dyn FnMut(&Path, bool) -> CargoResult<bool>,
     ) -> Option<CargoResult<Vec<PathBuf>>> {
```

Comment:
FWIW I know you aren't responsible for this but this return type seems pretty wonky nowadays and it may be best to switch the option/result so `?` and normal idiomatic error handling could be used here.

Reply:
Updated.